backend/src/main/java/com/dhbw/tutorsystem/security/authentication/AuthenticationController.java

```diff
@@ -0,0 +1,88 @@
+package com.dhbw.tutorsystem.security.authentication;
+
+import com.dhbw.tutorsystem.role.ERole;
+import com.dhbw.tutorsystem.role.Role;
+import com.dhbw.tutorsystem.role.RoleRepository;
+import com.dhbw.tutorsystem.security.jwt.JwtUtils;
+import com.dhbw.tutorsystem.security.services.UserDetailsImpl;
+import com.dhbw.tutorsystem.user.User;
+import com.dhbw.tutorsystem.user.UserRepository;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import javax.validation.Valid;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/authentication")
+public class AuthenticationController {
+    final AuthenticationManager authenticationManager;
+    final UserRepository userRepository;
+    final RoleRepository roleRepository;
+    final JwtUtils jwtUtils;
+    final PasswordEncoder encoder;
+
+    public AuthenticationController(AuthenticationManager authenticationManager, UserRepository userRepository,
+            RoleRepository roleRepository, JwtUtils jwtUtils, PasswordEncoder encoder) {
+        this.authenticationManager = authenticationManager;
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.jwtUtils = jwtUtils;
+        this.encoder = encoder;
+    }
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
+
+        Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(loginRequest.getEmail(), loginRequest.getPassword()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
```

Comment:
Please add a comment. What is the context and what is it used for?

Reply:
Fixed [400b1d0](https://github.com/FinnHuelsbusch/tutorensystem/pull/9/commits/400b1d013a14af47fc7e4d0b76474ee5c82c4b39)